src/vue/fields/SelectField.vue

```diff
@@ -18,59 +17,72 @@
       type="button"
       class="select-field__selected"
       :class="{
-        'select-field__selected--focused': isExpanded,
-        'select-field__selected--padding' : label
+        'select-field__selected--focused': isListOpened,
+        'select-field__selected--padding': label
       }"
       :disabled="disabled"
       @click.prevent="toggleListVisibility"
     >
       <span class="select-field__selected-value">
-        {{ getLabel(currentValue) || '&nbsp;' }}
+        {{ selected | getOptionText(keyAsValueText) }}
       </span>
       <i
         class="select-field__selected-icon mdi mdi-chevron-down"
-        :class="{ 'select-field__selected-icon--active': isExpanded }"
+        :class="{ 'select-field__selected-icon--active': isListOpened }"
       />
     </button>
     <div
       class="select-field__list"
       ref="list"
-      :class="{ 'select-field__list--active': isExpanded }"
+      :class="{ 'select-field__list--active': isListOpened }"
     >
       <button
-        v-for="(item, i) in values"
-        :key="i"
+        v-for="(item, index) in values"
+        :key="index"
         type="button"
         class="select-field__list-item"
-        :class="{
-          'select-field__list-item--selected':
-            getValue(selectedValue) === getValue(item)
-        }"
+        :class="{ 'select-field__list-item--selected': preSelected === item }"
         @click.prevent="selectItem(item)"
       >
-        {{ getLabel(item) }}
+        {{ item | getOptionText(keyAsValueText) }}
       </button>
     </div>
-    <p v-if="errorMessage" class="select-field__err-mes">
+    <p
+      v-if="errorMessage"
+      class="select-field__err-mes"
+    >
       {{ errorMessage }}
     </p>
+    <!-- Selected: {{ value }} -->
   </div>
 </template>
 
 <script>
 import { KEY_CODES } from '@/js/const/key-codes.const'
-import _isObject from 'lodash/isObject'
+import _get from 'lodash/get'
 
 const EVENTS = {
   input: 'input',
 }
 
 export default {
   name: 'select-field',
+
+  filters: {
+    getOptionText (item, keyAsValueText) {
+      const result = keyAsValueText
+        ? _get(item, keyAsValueText, item)
+        : item
+      return typeof result === 'function'
```

Comment:
pls gimme a doc on how to use the component

Reply:
too many complex stuff I don't want to understand

Reply:
the doc is ready